common/src/main/scala/package.scala

```diff
@@ -195,4 +223,18 @@ package object eoi {
     def dedupPreserveOrder: List[A] = collection.mutable.LinkedHashSet(in:_*).toList
   }
 
+  implicit class RichLocalDate(val value: LocalDate) extends AnyVal {
+
+    // we can't use TemporalAdjusters here because it's not supported in scalajs 0.6
+    def previousOrSame(day: java.time.DayOfWeek): LocalDate = {
+      val diff = ((value.getDayOfWeek.getValue + 7) - day.getValue) % 7
+      value.minusDays(diff)
+    }
+
+    def nextOrSame(day: java.time.DayOfWeek): LocalDate = {
```

Comment:
there are methods in the java which do this and the above already which are under `TemporalAdjusters`: https://stackoverflow.com/a/33839611

Reply:
Sadly they don't work with ScalaJS. I put an explanation of this in the comment a few lines above. 
I tried to keep the method names and structure as similar as possible for familiarity.

Reply:
oh, didn't spot that comment